keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/constraint/ImageExistsConstraintEvaluator.kt

```diff
@@ -41,25 +43,38 @@ class ImageExistsConstraintEvaluator(
     deliveryConfig: DeliveryConfig,
     targetEnvironment: Environment
   ): Boolean =
-    artifact !is DebianArtifact || imagesExistInAllRegions(version, artifact.vmOptions)
+    artifact !is DebianArtifact || imagesExistInAllRegions(version, artifact.vmOptions, artifact)
 
-  private fun imagesExistInAllRegions(version: String, vmOptions: VirtualMachineOptions): Boolean =
-    runBlocking {
+  /**
+   * Check both clouddriver cached images and the images we've baked that haven't been
+   * cached yet.
+   */
+  private fun imagesExistInAllRegions(version: String, vmOptions: VirtualMachineOptions, artifact: DebianArtifact): Boolean {
+    val clouddriverImages = runBlocking {
       imageService.getLatestNamedImages(
         appVersion = version.parseAppVersion(),
         account = defaultImageAccount,
         regions = vmOptions.regions
       )
     }
-      .also {
-        if (it.keys.containsAll(vmOptions.regions)) {
-          log.info("Found AMIs for all desired regions for {}", version)
-        } else {
-          log.warn("Missing regions {} for {}", (vmOptions.regions - it.keys).sorted().joinToString(), version)
-          eventPublisher.publishEvent(MissingRegionsDetected(version))
-        }
-      }
-      .keys.containsAll(vmOptions.regions)
+
+    val bakedImage = bakedImageRepository.getLatestByArtfiactVerstion(version, artifact)
+
+    if (clouddriverImages.keys.containsAll(vmOptions.regions)) {
+      log.info("Found AMIs for all desired regions in clouddriver cache for {}", version)
+    } else {
+      log.warn("Missing regions {} clouddriver cache for {}", (vmOptions.regions - clouddriverImages.keys).sorted().joinToString(), version)
+      eventPublisher.publishEvent(MissingRegionsDetected(version))
+    }
+
+    if (!clouddriverImages.keys.containsAll(vmOptions.regions) && bakedImage != null) {
```

Comment:
The first part of this condition is the same as the `else` condition above, so you could nest an `if (bakedImage != null)` there instead to simplify.